snappi_ixnetwork/trafficitem.py

```diff
@@ -690,10 +738,26 @@ def results(self, request):
             msg = "Invalid format of flow_names passed {},\
                     expected list".format(flow_names)
             raise Exception(msg)
-
-        filter = {'property': 'name', 'regex': '.*'}
-        filter['regex'] = '^(%s)$' % '|'.join(
-                        self._api.special_char(flow_names))
+        final_flow_names = [
+            f.name for f in self._api._config.flows
+            if f.metrics.enable is True and f.name in flow_names
+        ]
+        if final_flow_names == []:
+            msg = """
+            To fetch flow metrics at least one flow shall have metric enabled
+            """
+            raise Exception(msg.strip())
+        diff = set(flow_names).difference(final_flow_names)
+        if len(diff) > 0:
+            self._api.warning(
```

Comment:
We shouldn't have to throw warnings here.
- If flow_names=[], we should only return flow metrics for flows where metrics is enabled
- If flow_names contains name of at least one flow where metrics is not enabled, we should raise error

Error message should say:
`Flow metrics is not enabled on flow %s`